### PR TITLE
Fixes wrong toast message while deleting saved/blank forms

### LINKED ISSUE #539: Wrong display message while deleting saved/blank forms
#### Software and hardware versions 
Collect v1.4.16, Android v4.4.2, API 19

#### Problem description
Deleting saved forms or blank forms from `Delete Saved Forms` option on `Main Menu` displays wrong Toast message (Selected forms failed to delete). But the saved form or the blank form actually gets deleted

#### Steps to reproduce the problem
1. Select `Fill Blank Form` 
2. Fill any form and click the `save` icon
3. Save the changes and go back to main menu
4. Click on `Delete Saved Form`
5. Delete the `saved form(s)` or `blank form(s)`

#### Expected behavior
Display message "1 form(s) successfully deleted!"

### Actual behaviour 
Message displayed "Sorry, -1 of 0 selected form(s) failed to delete!"

device-2017-03-08-104822

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #539

### CODE REVIEW

**Comment by lognaturel** [comment]
There was a recent change to using listView state directly to know how many items were checked rather than relying on a separate counter and that's what caused the problem, right? When items were deleted, the checked item count went to 0. Am I understanding that correctly?

I think this is the right fix but I don't think the delete tasks should know anything about the concept of "checked" items. Instead of `checkedCount` and `getCheckedCount`, I'd recommend something like `toDeleteCount`. How does that sound?

**Comment by shobhitagarwal1612** [comment]
Yes you got it right! The scene is that as soon as the instances were deleted, the adapter was updated asynchronously and the new size becomes zero. So I passed the `checkedCount` to the AsyncTask to handle it accordingly. Now that you mention, it does seem useless for the delete task to know anything about the `checkedCount`. @lognaturel Your proposal does sounds good. So should I rename the variables to `toDeleteCount` and `getDeleteCount`?

**Comment by lognaturel** [comment]
@shobhitagarwal1612 Ok, thanks, I just wanted to make sure we were on the same page. :) Yes, please change the names.

**Comment by shobhitagarwal1612** [comment]
@lognaturel Done

File: collect_app/src/main/java/org/odk/collect/android/activities/DataManagerList.java [hunk 1]
```diff
@@ -235,7 +235,8 @@ public void deleteComplete(int deletedInstances) {
         Log.i(t, "Delete instances complete");
         logger.logAction(this, "deleteComplete",
                 Integer.toString(deletedInstances));
-        final int checkedCount = getCheckedCount();
+        final int checkedCount = mDeleteInstancesTask.getToDeleteCount();
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/DataManagerList.java]: Doesn't it make sense for this local variable to also be `toDeleteCount`?

File: collect_app/src/main/java/org/odk/collect/android/activities/FormManagerList.java [hunk 2]
```diff
@@ -265,12 +264,14 @@ public void syncComplete(String result) {
     public void deleteComplete(int deletedForms) {
         Log.i(t, "Delete forms complete");
         logger.logAction(this, "deleteComplete", Integer.toString(deletedForms));
-        if (deletedForms == getCheckedCount()) {
+        final int checkedCount = mBackgroundTasks.mDeleteFormsTask.getToDeleteCount();
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormManagerList.java]: Same comment as above. By the time we're here, it's no longer the checked count even from the lists' perspective. It truly represents the count of stuff we previously meant to delete.

File: collect_app/src/main/java/org/odk/collect/android/activities/DataManagerList.java (hunk 1 above)
Comment (shobhitagarwal1612) [reply collect_app/src/main/java/org/odk/collect/android/activities/DataManagerList.java]: yes it does.

**Pull request merged**